frontend/class-frontend.php

```diff
@@ -31,11 +36,46 @@ class WPSEO_Frontend {
 		private $ob_started = false;
 
 		/**
+		 * Holds the canonical URL for the current page
+		 *
+		 * @var string
+		 */
+		private $canonical = null;
+
+		/**
+		 * Holds the canonical URL for the current page that cannot be overriden by a manual canonical input
+		 *
+		 * @var string
+		 */
+		private $canonical_no_override = null;
+
+		/**
+		 * Holds the canonical URL for the current page without pagination
+		 *
+		 * @var string
+		 */
+		private $canonical_unpaged = null;
+
+		/**
+		 * Holds the pages meta description
+		 *
+		 * @var string
+		 */
+		private $metadesc = null;
+
+		/**
+		 * Holds the generated title for the page
+		 *
+		 * @var string
+		 */
+		private $title = null;
+
+		/**
 		 * Class constructor
 		 *
 		 * Adds and removes a lot of filters.
 		 */
-		function __construct() {
+		public function __construct() {
```

Comment:
A singleton resource should only be reachable through it's static creation method (in this case `get_singleton`) to prevent more than one instance being created. Please make the constructor `protected`. More info -> http://www.phptherightway.com/pages/Design-Patterns.html